### PR TITLE
chore: remove hacks in cosmwasm consumer client

### PR DESCRIPTION
**Pull request by gusin13**

### CODE REVIEW

File: clientcontroller/cosmwasm/consumer.go [hunk 1]
```diff
@@ -590,6 +572,7 @@ func (wc *CosmwasmConsumerController) queryCometBestBlock() (*fptypes.BlockInfo,
 	}, nil
 }
 
+//nolint:unused
```
Comment (bap2pecs) [inline clientcontroller/cosmwasm/consumer.go:575]: does lint complain b/c of build tags?

File: itest/cosmwasm/bcd/bcd_test_manager.go [hunk 2]
```diff
@@ -83,6 +83,8 @@ func StartBcdTestManager(t *testing.T) *BcdTestManager {
 	bc, err := bbncc.NewBabylonController(cfg.BabylonConfig, &cfg.BTCNetParams, logger)
 	require.NoError(t, err)
 
+	cfg.FastSyncInterval = 0 // disable fast sync
```
Comment (bap2pecs) [inline itest/cosmwasm/bcd/bcd_test_manager.go:86]: why this is disabled? did you meet w any issues as we did in https://github.com/babylonchain/finality-provider/pull/484?

Comment (bap2pecs) [reply itest/cosmwasm/bcd/bcd_test_manager.go:86]: i see. feel free to check if there is anything you can reuse from our tests. iirc, we have some helper functions to wait for delegation to be active

**Review by SebastianElvis — APPROVED** [review]
Lgtm, great work on getting rid of all hacks!

**Comment by SebastianElvis** [comment]
I realised a strong argument that we should stick to the hacks in the two queries. It's possible that the consuemr system only wants to integrate with 0.3.0 that has finality signature submissions but not 0.4.0 that has on-chain finality round, as  0.4.0 needs breaking chanegs in their code base. If we query indexed blocks in the contract then the FP won't work with 0.3.0. If we query Comet blocks then there is no such problem

**Review by maurolacy — APPROVED** [review]
LGTM.

Will try this in the deployment repo next.

**Comment by maurolacy** [comment]
> It's possible that the consuemr system only wants to integrate with 0.3.0 that has finality signature submissions but not 0.4.0 that has on-chain finality round, as 0.4.0 needs breaking chanegs in their code base.

I would say, they should attempt those changes, sooner that later. I would go for the 'latest and greatest' 0.4.0 directly.

**Comment by maurolacy** [comment]
After offline talks, we can release this in time (after it works) as a 0.4.1 release.

**Comment by SebastianElvis** [comment]
As per yesterday's discussion, probably let's simply query Cosmos SDK blocks rather than indexed blocks such that FP remains compatible with chains with or without Babylon SDK?

**Comment by maurolacy** [comment]
Closing without merging, after discussion.

**Comment by maurolacy** [comment]
Feel free to re-open if some changes here are to be merged.

**Pull request closed without merging**